tests/workflows/integration_tests/execution/test_workflow_with_gaze.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import pytest
+import sys
+
+from inference.core.env import WORKFLOWS_MAX_CONCURRENT_STEPS
+from inference.core.managers.base import ModelManager
+from inference.core.workflows.core_steps.common.entities import StepExecutionMode
+from inference.core.workflows.execution_engine.core import ExecutionEngine
+from tests.workflows.integration_tests.execution.workflows_gallery_collector.decorators import (
+    add_to_workflows_gallery,
+)
+
+GAZE_DETECTION_WORKFLOW = {
+    "version": "1.0",
+    "inputs": [
+        {"type": "WorkflowImage", "name": "image"},
+        {"type": "WorkflowParameter", "name": "do_run_face_detection", "default_value": True},
+    ],
+    "steps": [
+        {
+            "type": "roboflow_core/gaze@v1",
+            "name": "gaze",
+            "images": "$inputs.image",
+            "do_run_face_detection": "$inputs.do_run_face_detection",
+        },
+        {
+            "type": "roboflow_core/keypoint_visualization@v1",
+            "name": "visualization",
+            "predictions": "$steps.gaze.face_predictions",
+            "image": "$inputs.image",
+            "annotator_type": "vertex",
+            "color": "#A351FB",
+            "text_color": "black",
+            "text_scale": 0.5,
+            "text_thickness": 1,
+            "text_padding": 10,
+            "thickness": 2,
+            "radius": 10,
+        },
+    ],
+    "outputs": [
+        {
+            "type": "JsonField",
+            "name": "face_predictions",
+            "selector": "$steps.gaze.face_predictions",
+        },
+        {
+            "type": "JsonField",
+            "name": "yaw_degrees",
+            "selector": "$steps.gaze.yaw_degrees",
+        },
+        {
+            "type": "JsonField",
+            "name": "pitch_degrees",
+            "selector": "$steps.gaze.pitch_degrees",
+        },
+        {
+            "type": "JsonField",
+            "name": "visualization",
+            "selector": "$steps.visualization.image",
+        },
+    ],
+}
+
+
+@add_to_workflows_gallery(
+    category="Workflows with foundation models",
+    use_case_title="Gaze Detection Workflow",
+    use_case_description="""
+This workflow uses L2CS-Net to detect faces and estimate their gaze direction.
+The output includes:
+- Face detections with facial landmarks
+- Gaze angles (yaw and pitch) in degrees
+- Visualization of facial landmarks
+""",
+    workflow_definition=GAZE_DETECTION_WORKFLOW,
+    workflow_name_in_app="gaze-detection",
+)
+@pytest.mark.skip(reason="Test not supported on Python 3.12+, skipping due to dependencies conflict when building CI")
```

Comment:
Dependencies conflict when installing gaze on python 3.12 needs to be resolved and this test should then be reenabled.